dulwich/tests/test_graph.py

```diff
@@ -32,9 +32,12 @@ def run_test(dag, inputs):
         def lookup_parents(commit_id):
             return dag[commit_id]
 
+        def lookup_stamp(commit_id):
+            return 100
```

Comment:
Why 100?

Reply:
Your graph as used in tests have no commit time info, just parentage.  Since the more efficient version needs commit time info, it must be supplied by the test itself.  A constant commit time is returned for all points causing the test to be equivalent to its initial form.  Any constant value will work.

Reply:
It acts as the regression test you asked for above as it reverts the test to exactly the behaviour previously tested.

Reply:
please add a comment to that effect

Reply:
Will do.

Reply:
fixed in the updated PR